### PR TITLE
feat: Add Missing Methods for Execution details

### LINKED ISSUE #18: Add Missing Methods for Execution details
Some endpoints of the following section can't be hit from this SDK. 

We would like to ensure that all endpoints can be hit. For this issue, we would like to build out the Execution details

Missing methods:

Get Execution Details - https://docs.novu.co/api-reference/execution-details/get-execution-details

### PR DESCRIPTION
**Pull request by sccalabr**
Resolves #18

### CODE REVIEW

**Comment by doomspork** [comment]
@sccalabr could you please address the failing CI checks?

**Comment by unicodeveloper** [comment]
@sccalabr Please attend to the failing tests.

File: test/novu/execution_details_test.exs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    on_exit(fn ->
+      Application.put_env(:novu, :domain, current_domain)
+    end)
+
+    {:ok, bypass: bypass}
+  end
+
+  describe "get-execution-details/1" do
+    test "creates a GET to /v1/execution-details", %{bypass: bypass} do
+      notification_group_name = "Test"
```
Comment (btkostner) [inline test/novu/execution_details_test.exs:27]: This variable is not used. Please remove it.

**Review by btkostner — CHANGES_REQUESTED** [review]

**Comment by unicodeveloper** [comment]
@sccalabr All the CI tests are failing. Please look into it

**Comment by btkostner** [comment]
The CI failures look to be coming from main branch. I'll open a PR soon to fix them

**Review by btkostner — CHANGES_REQUESTED** [review]
Code looks good. Just need to fix the [format error](https://github.com/novuhq/novu-elixir/actions/runs/6570062070/job/17875287484#step:6:13) in CI by running `mix format` and we should be good.

**Review by doomspork — APPROVED** [review]

**Pull request merged**